### PR TITLE
Feature/uia permission tests

### PR DESCRIPTION
**Pull request by nlesniewski**

### CODE REVIEW

File: app/src/androidTest/java/com/invisibleteam/goinvisible/Config.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.invisibleteam.goinvisible;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/Config.java]: Please remove this

File: app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.invisibleteam.goinvisible;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.uiautomator.UiDevice;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java]: Please remove this

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.invisibleteam.goinvisible.pages;
+
+import android.support.test.uiautomator.By;
+import android.support.test.uiautomator.BySelector;
+import android.support.test.uiautomator.Until;
+
+import com.invisibleteam.goinvisible.Config;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java]: Please remove this

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+public class FileView extends UiDialog {
+
+    // Main activity selectors
+    private static final BySelector mainActivitySelector = By.res(Config.GOINVISIBLE_PACKAGE, "activity_main");
+    private static final BySelector noImagesInformationSelector = By.res(Config.GOINVISIBLE_PACKAGE, "no_images_information");
+
+    public static Boolean isOpened() {
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java]: I understand that isOpened concerns FileView but what FileView is?

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.invisibleteam.goinvisible.pages;
+
+import android.support.test.uiautomator.By;
+import android.support.test.uiautomator.BySelector;
+import android.support.test.uiautomator.Until;
+
+import com.invisibleteam.goinvisible.Config;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java]: Please remove this

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+public class MissingPermissionsDialog extends UiDialog {
+
+    // Missing permissions dialog selectors
+    private static final BySelector missingPermissionsTitleSelector = By.res(GOINVISIBLE_PACKAGE, "alertTitle");
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java]: Please write all constants with Upper Case (MISSING_PERMISSIONS_TITLE_SELECTOR)

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.invisibleteam.goinvisible.Config;
+import com.invisibleteam.goinvisible.UiDeviceProvider;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
+ */
+public class UiDialog {
+    @SuppressWarnings("SpellCheckingInspection")
+    protected static String GOINVISIBLE_PACKAGE = Config.GOINVISIBLE_PACKAGE;
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: final

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    protected static int TIMEOUT = Config.LAUNCH_TIMEOUT;
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: final

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+    protected static UiDevice mDevice = UiDeviceProvider.getInstance();
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: final

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/system/SystemPermissionDialog.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import android.support.test.uiautomator.BySelector;
+import android.support.test.uiautomator.UiObject2;
+import android.support.test.uiautomator.Until;
+
+import com.invisibleteam.goinvisible.pages.UiDialog;
+
+import static com.invisibleteam.goinvisible.Config.LAUNCH_TIMEOUT;
+
+/**
+ * Created by nlesniewski on 14.02.2017.
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/pages/system/SystemPermissionDialog.java]: Please remove this

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java (hunk 4 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java]: FileView  is the list of images.

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/system/SystemPermissionDialog.java (hunk 10 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/system/SystemPermissionDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java (hunk 5 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java (hunk 3 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java (hunk 2 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/Config.java (hunk 1 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/Config.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java (hunk 4 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/FileView.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java (hunk 6 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/MissingPermissionsDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java (hunk 7 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java (hunk 8 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java (hunk 9 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/pages/UiDialog.java]: :white_check_mark:

File: app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package com.invisibleteam.goinvisible;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.uiautomator.UiDevice;
+
+public class UiDeviceProvider {
+    private static UiDevice ourInstance = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
```
Comment (BartSoftware) [inline app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java]: maybe applicationInstance/appInstance is a better name?

**Review by BartSoftware — DISMISSED** [review]

**Review by kejmil01 — APPROVED** [review]

File: app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java (hunk 11 above)
Comment (nlesniewski) [reply app/src/androidTest/java/com/invisibleteam/goinvisible/UiDeviceProvider.java]: Didn't bothered at all:
- it was automatically generated
- it is private
- whole class is 15 lines of code and no functionality

It is not even a trivial

**Review by michal1990 — APPROVED** [review]

**Pull request merged**